In [2]:
caminho <- c("..", "dados", "Data Sets- R")

## C2

In [8]:
filepath = append(caminho, "lawsch85.RData")
filepath = paste(filepath, collapse='/')
filepath

[1] "../dados/Data Sets- R/lawsch85.RData"

In [9]:
load(filepath)

In [26]:
regr1 = lm(log(salary) ~ LSAT + GPA + log(libvol) + log(cost) + rank, data=data)
summary(regr1)


Call:
lm(formula = log(salary) ~ LSAT + GPA + log(libvol) + log(cost) + 
    rank, data = data)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.301356 -0.084982 -0.004359  0.077935  0.288614 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  8.3432262  0.5325192  15.667  < 2e-16 ***
LSAT         0.0046965  0.0040105   1.171  0.24372    
GPA          0.2475238  0.0900371   2.749  0.00683 ** 
log(libvol)  0.0949932  0.0332544   2.857  0.00499 ** 
log(cost)    0.0375539  0.0321061   1.170  0.24427    
rank        -0.0033246  0.0003485  -9.541  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1124 on 130 degrees of freedom
  (20 observations deleted due to missingness)
Multiple R-squared:  0.8417,	Adjusted R-squared:  0.8356 
F-statistic: 138.2 on 5 and 130 DF,  p-value: < 2.2e-16


In [27]:
regr2 = lm(log(salary) ~ rank, data=data)
summary(regr2)


Call:
lm(formula = log(salary) ~ rank, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.36959 -0.09168 -0.02628  0.10016  0.36529 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.9379218  0.0194867  561.30   <2e-16 ***
rank        -0.0049415  0.0002014  -24.54   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1227 on 146 degrees of freedom
  (8 observations deleted due to missingness)
Multiple R-squared:  0.8048,	Adjusted R-squared:  0.8035 
F-statistic:   602 on 1 and 146 DF,  p-value: < 2.2e-16


In [40]:
regr3 = lm(log(salary) ~ LSAT + GPA + log(libvol) + log(cost) + rank + log(clsize) + log(faculty), data=data)
summary(regr3)


Call:
lm(formula = log(salary) ~ LSAT + GPA + log(libvol) + log(cost) + 
    rank + log(clsize) + log(faculty), data = data)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.299228 -0.085435 -0.001876  0.077239  0.273140 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)   8.2599386  0.5847660  14.125  < 2e-16 ***
LSAT          0.0052982  0.0042529   1.246  0.21522    
GPA           0.2562574  0.0956956   2.678  0.00842 ** 
log(libvol)   0.0714880  0.0415489   1.721  0.08784 .  
log(cost)     0.0389202  0.0352609   1.104  0.27184    
rank         -0.0033969  0.0003604  -9.425 3.32e-16 ***
log(clsize)   0.0183748  0.0380339   0.483  0.62987    
log(faculty) -0.0027450  0.0321624  -0.085  0.93212    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1138 on 123 degrees of freedom
  (25 observations deleted due to missingness)
Multiple R-squared:  0.842,	Adjusted R-squared:  0.833 
F-statistic:

## i
H0: Brank = 0
O valor do p-value para rank é muito baixo, em qualquer equação. Portanto, rank é estatisticamente 
significante e devemos rejeitar a hipótese nula.

In [30]:
# install.packages("car")
library(car)
linearHypothesis(regr1, c("rank = 0"))

Res.Df RSS      Df Sum of Sq F        Pr(>F)      
1 131    2.792977 NA       NA        NA           NA
2 130    1.642730  1 1.150247  91.02661 1.121018e-16

## ii
H0: bLSAT, bGPA = 0

Temos em regr1:
* LSAT         0.0046965  0.0040105   1.171  0.24372   
* GPA          0.2475238  0.0900371   2.749  0.00683 ** 

Portanto, GPA é significante, mas LSAT não é.

Em regr4 testou-se LSAT:GPA:
* LSAT        -0.1424654  0.0309557  -4.602 9.87e-06 ***
* GPA         -6.9187233  1.4988798  -4.616 9.33e-06 ***
* LSAT:GPA     0.0446301  0.0093203   4.788 4.53e-06 ***

O que inverteu os sinais de bLSAT e bGPA. De qualquer forma, a significância aumentou.

LSAT e GPA parecem correlacionados

Rodar um f_test confirma a relevância de ambos (abaixo)

Também foi rodada uma regressão apenas com LSAT(regr5), o que confirma relevância
* LSAT         0.0112920  0.0032932   3.429 0.000811 ******

In [32]:
linearHypothesis(regr1, c("LSAT = 0", "GPA = 0"))

Res.Df RSS      Df Sum of Sq F       Pr(>F)      
1 132    1.894238 NA        NA      NA           NA
2 130    1.642730  2 0.2515081 9.95174 9.518231e-05

In [34]:
regr4 = lm(log(salary) ~ LSAT*GPA + log(libvol) + log(cost) + rank, data=data)
summary(regr4)


Call:
lm(formula = log(salary) ~ LSAT * GPA + log(libvol) + log(cost) + 
    rank, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.30163 -0.07267 -0.01317  0.05786  0.32407 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 32.5043916  5.0696878   6.412 2.49e-09 ***
LSAT        -0.1424654  0.0309557  -4.602 9.87e-06 ***
GPA         -6.9187233  1.4988798  -4.616 9.33e-06 ***
log(libvol)  0.0549952  0.0318749   1.725   0.0869 .  
log(cost)    0.0048934  0.0304719   0.161   0.8727    
rank        -0.0036289  0.0003285 -11.046  < 2e-16 ***
LSAT:GPA     0.0446301  0.0093203   4.788 4.53e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.104 on 129 degrees of freedom
  (20 observations deleted due to missingness)
Multiple R-squared:  0.8656,	Adjusted R-squared:  0.8593 
F-statistic: 138.4 on 6 and 129 DF,  p-value: < 2.2e-16


In [38]:
regr5 = lm(log(salary) ~ LSAT + log(libvol) + log(cost) + rank, data=data)
summary(regr5)


Call:
lm(formula = log(salary) ~ LSAT + log(libvol) + log(cost) + rank, 
    data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.29908 -0.09323 -0.01196  0.07840  0.26839 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  8.4632252  0.5438489  15.562  < 2e-16 ***
LSAT         0.0112920  0.0032932   3.429 0.000811 ***
log(libvol)  0.0987489  0.0340478   2.900 0.004374 ** 
log(cost)    0.0014094  0.0300148   0.047 0.962619    
rank        -0.0036816  0.0003314 -11.111  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1152 on 131 degrees of freedom
  (20 observations deleted due to missingness)
Multiple R-squared:  0.8325,	Adjusted R-squared:  0.8274 
F-statistic: 162.8 on 4 and 131 DF,  p-value: < 2.2e-16


## iii

H0: bclsize, bfaculty = 0

Em regr3:
* log(clsize)   0.0183748  0.0380339   0.483  0.62987    
* log(faculty) -0.0027450  0.0321624  -0.085  0.93212

Por si só, os t-value e p-value demonstram não significância.

O f-test, abaixo, mostra a falta de significância conjunta.

In [41]:
linearHypothesis(regr3, c("log(clsize) = 0", "log(faculty) = 0"))

Res.Df RSS      Df Sum of Sq   F         Pr(>F)   
1 125    1.597422 NA          NA        NA        NA
2 123    1.593476  2 0.003945688 0.1522833 0.8589066